frontend/__tests__/unit/components/Badges.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { render, screen } from '@testing-library/react'
+import React from 'react'
+import Badges from 'components/Badges'
+
+// Mock FontAwesome
+jest.mock('@fortawesome/fontawesome-svg-core', () => ({
+  findIconDefinition: jest.fn(({ iconName }: { iconName: string }) => {
+    // Return truthy value for valid icons, null for invalid
+    return iconName === 'medal' ? { iconName } : null
+  }),
+}))
+
+jest.mock('wrappers/FontAwesomeIconWrapper', () => {
+  return function MockFontAwesomeIconWrapper({ icon }: { icon: string }) {
+    const iconName = icon.split(' ').pop()?.replace('fa-', '') || icon.replace('fa-', '')
+    return <span data-testid={`icon-${iconName}`} />
+  }
+})
+
+jest.mock('@heroui/tooltip', () => ({
+  Tooltip: ({
+    children,
+    content,
+    isDisabled,
+  }: {
+    children: React.ReactNode
+    content: string
+    isDisabled?: boolean
+  }) => {
+    if (isDisabled) {
+      return <>{children}</>
+    }
+    return (
+      <div data-testid="tooltip" data-content={content}>
+        {children}
+      </div>
+    )
+  },
+}))
+
+describe('Badges Component', () => {
+  const defaultProps = {
+    name: 'Test Badge',
+    cssClass: 'fa-solid fa-medal',
+  }
+
+  it('renders valid icon with tooltip', () => {
+    render(<Badges {...defaultProps} />)
+
+    expect(screen.getByTestId('icon-medal')).toBeInTheDocument()
```

Comment:
This tests the mock's behavior (rendering a span with test ID) rather than the real component's behavior - so also, never caught the actual error.